### PR TITLE
Make unknown command and internal error messages customisable

### PR DESCRIPTION
**Pull request by VytskaLT**
Makes the unknown command and internal error messages customisable in PlayerCommandPreprocessEvent. Useful for multi-language servers.

### CODE REVIEW

File: patches/api/0101-Make-unknown-and-internal-command-messages-customisa.patch [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
++        return unknownCommandMessage;
++    }
++
++    /**
++     * Sets the message that the player will receive if the command wasn't found.
++     *
++     * @param unknownCommandMessage new message that the player will receive if the command wasn't found. null if no message should be sent.
++     */
++    public void setUnknownCommandMessage(String unknownCommandMessage) {
++        this.unknownCommandMessage = unknownCommandMessage;
```
Comment (AustinLMayes) [inline patches/api/0101-Make-unknown-and-internal-command-messages-customisa.patch:53]: This needs null and empty validation.

File: patches/api/0101-Make-unknown-and-internal-command-messages-customisa.patch [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
++        return internalErrorMessage;
++    }
++
++    /**
++     * Sets the message that the player will receive if an internal error occurred.
++     *
++     * @param internalErrorMessage new message that the player will receive if an internal error occurred. null if no message should be sent.
++     */
++    public void setInternalErrorMessage(String internalErrorMessage) {
++        this.internalErrorMessage = internalErrorMessage;
```
Comment (AustinLMayes) [inline patches/api/0101-Make-unknown-and-internal-command-messages-customisa.patch:71]: This needs null and empty validation.

File: patches/server/0193-Make-unknown-and-internal-command-messages-customisa.patch [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -619,8 +619,14 @@ public final class CraftServer implements Server {
+         }
+     }
+ 
++    // SportPaper start
+     @Override
+-    public boolean dispatchCommand(CommandSender sender, String commandLine) {
++    public boolean dispatchCommand(CommandSender sender, String commandLine) throws CommandException {
```
Comment (AustinLMayes) [inline patches/server/0193-Make-unknown-and-internal-command-messages-customisa.patch:54]: Don't need the `throws`

File: patches/server/0193-Make-unknown-and-internal-command-messages-customisa.patch [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
++++ b/src/main/java/net/minecraft/server/PlayerConnection.java
+@@ -1319,7 +1319,10 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
+ 
+         CraftPlayer player = this.getPlayer();
+ 
+-        PlayerCommandPreprocessEvent event = new PlayerCommandPreprocessEvent(player, s, new LazyPlayerSet());
++        // SportPaper - make unknown command and internal error messages customisable
++        PlayerCommandPreprocessEvent event = new PlayerCommandPreprocessEvent(player, s, new LazyPlayerSet(),
++                org.spigotmc.SpigotConfig.unknownCommandMessage,
++                org.bukkit.ChatColor.RED + "An internal error occurred while attempting to perform this command");
```
Comment (AustinLMayes) [inline patches/server/0193-Make-unknown-and-internal-command-messages-customisa.patch:20]: Make this configurable via config as well

**Review by AustinLMayes — CHANGES_REQUESTED** [review]

**Comment by VytskaLT** [comment]
Can this be reviewed?

**Comment by Electroid** [comment]
So this is kinda off-spec with newer Spigot/Paper, but I think its a fine addition. GTM

**Pull request merged**